iaso/models/stocks.py

```diff
@@ -0,0 +1,306 @@
+        return self.filter_for_project(project)
+
+    def filter_for_project(self, project: Project):
+        return self.filter(
+            sku__account=project.account,
+            sku__projects__in=[project],
+        )
+
+
+class StockLedgerItem(models.Model):
+    """
+    An action that changed the stock item value based on a StockItemRule.
+
+    Note: We are not referencing StockItemRule as they can change over time but a StockLedgerItem is immutable.
+    """
+
+    id = models.UUIDField(primary_key=True, auto_created=True, default=uuid.uuid4, editable=False)
+    sku = models.ForeignKey(
+        "StockKeepingUnit", related_name="ledger_item_sku", null=False, blank=False, on_delete=models.CASCADE
+    )
+    org_unit = models.ForeignKey(
+        "OrgUnit", related_name="ledger_item_org_unit", null=False, blank=False, on_delete=models.CASCADE
+    )
+    rule = models.ForeignKey(
+        "StockItemRule", related_name="ledger_item_rule", null=True, blank=True, on_delete=models.CASCADE
+    )
+    submission = models.ForeignKey(
+        "Instance", related_name="ledger_item_instance", null=True, blank=True, on_delete=models.CASCADE
+    )
+    question = models.CharField(max_length=255, null=True, blank=True)
+    value = models.IntegerField()
+    impact = models.CharField(max_length=8, choices=StockImpacts.choices, null=True, blank=True)
+    created_by = models.ForeignKey(
+        User, related_name="ledger_item_created_by", null=False, blank=False, on_delete=models.CASCADE
+    )
+    created_at = models.DateTimeField()
+    received_at = models.DateTimeField(auto_now_add=True)
+
+    objects = models.Manager.from_queryset(StockLedgerItemQuerySet)()
+
+
+class StockItemRuleQuerySet(QuerySet):
+    def filter_for_user(self, user: typing.Optional[typing.Union[User, AnonymousUser]]):
+        if not user or not user.is_authenticated:
+            raise UserNotAuthError("User not Authenticated")
+
+        profile = user.iaso_profile
+        return self.filter(sku__account=profile.account)
+
+    def filter_for_user_and_app_id(self, user: typing.Optional[typing.Union[User, AnonymousUser]], app_id: str):
+        project = Project.objects.get_for_user_and_app_id(user, app_id)
+        return self.filter_for_project(project)
+
+    def filter_for_project(self, project: Project):
+        return self.filter(
+            sku__projects__in=[project],
+            form__projects__in=[project],
+        )
+
+
+class StockItemRule(models.Model):
+    """
+    Defines how a form impacts stock items.
+    """
+
+    sku = models.ForeignKey(
+        "StockKeepingUnit", related_name="stock_item_rule_sku", null=False, blank=False, on_delete=models.CASCADE
+    )
+    form = models.ForeignKey(
+        "Form", related_name="stock_item_rule_form", null=False, blank=False, on_delete=models.CASCADE
+    )
+    question = models.CharField(max_length=255, null=False, blank=False)
```

Comment:
Same comment about `null=True` and `blank=True` in `CharField`
```suggestion
    question = models.CharField(max_length=255, blank=False)
```